src/runtime_src/hip/core/memory_pool.cpp

```diff
@@ -0,0 +1,407 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifdef _WIN32
+#include <Windows.h>
+#include <Memoryapi.h>
+#else
+#include <sys/mman.h>
+#endif
+
+#include "core/common/unistd.h"
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+
+#include "common.h"
+#include "memory_pool.h"
+
+namespace xrt::core::hip
+{
+  // Global map of memory_pool associated with device id.
+  std::map<int, std::list<std::shared_ptr<memory_pool>>> memory_pool_db;
+  
+  // Global map of memory_pool associated with its handle.
+  xrt_core::handle_map<mem_pool_handle, std::shared_ptr<memory_pool>> mem_pool_cache;
+
+  static void
+  dlinkedlist_delete(std::shared_ptr<memory_pool_slot>& head, std::shared_ptr<memory_pool_slot>& node)
+  {
+    if (node->m_prev == nullptr) {
+      head = node->m_next;
+      if (node->m_next != nullptr) node->m_next->m_prev = nullptr;
+    } else {
+      node->m_prev->m_next = node->m_next;
+      if (node->m_next!= nullptr) {
+        node->m_next->m_prev = node->m_prev;
+      }
+    }
+  }
+
+  static void
+  dlinkedlist_insert(std::shared_ptr<memory_pool_slot>& head, std::shared_ptr<memory_pool_slot>& node)
+  {    
+    node->m_prev = nullptr;
+    node->m_next = head;
+    if (head != nullptr) {
+      head->m_prev = node;
+    }
+    head = node;
```

Comment:
```suggestion
    if (head != nullptr)
      head->m_prev = node;
    
    head = node;
```
Nit.  Just what most code does, no curly braces for single statement, but new line after

Reply:
fixed.